tdms/include/field.h

```diff
@@ -2,14 +2,44 @@
 #include "simulation_parameters.h"
 
 
+/**
+ * A generic grid entity. For example:
+ *
+ *      _____________________ (2, 1)
+ *     /           /         /
+ *    /                     /
+ *   /___________/_________/
+ *  (0, 0)     (1, 0)     (2, 0)
+ *
+ *  has I_tot = 2, J_tot = 1, K_tot = 0.
+ */
+class Grid{
+
+protected:
+    int I_tot = 0;
+    int J_tot = 0;
+    int K_tot = 0;
+
+    /**
+     * Is this grid 0 x 0 x 0?
+     * @return True if there are no components
+     */
+    bool has_no_elements() const {return I_tot == 0 && J_tot == 0 && K_tot == 0;};
+};
+
+
 struct xyz_arrays{
     double ***x;
     double ***y;
     double ***z;
 };
 
 
-class Field{
+/**
+ * A complex field defined over a grid. Has real and imaginary components
```

Comment:
```suggestion
 * A complex vector field defined over a grid. Has three real and imaginary vector components
```
If I'm interpreting the code correctly?